ecmaScript.js

```diff
@@ -10,24 +10,24 @@ function parseLiteral (node, context) {
   throw new Error(`Cannot load ecmaScript code from node ${node}`)
 }
 
-function parseNamedNode (node, dataset, { context }) {
+function parseNamedNode (node, dataset, { basePath }) {
   const def = cf(dataset)
   const cfNode = def.node(node)
   const link = cfNode.out(code('link'))
 
   if (link.term && link.term.termType === 'NamedNode') {
-    return iriRequire(link.value, context.basePath)
+    return iriRequire(link.value, basePath)
   }
 
   throw new Error(`Cannot load ecmaScript code from node ${node}`)
 }
 
-function loader (node, dataset, { context }) {
+function loader (node, dataset, options) {
   if (node && node.termType === 'Literal') {
-    return parseLiteral(node, context)
+    return parseLiteral(node, options)
   }
 
-  return parseNamedNode(node, dataset, { context })
+  return parseNamedNode(node, dataset, options)
```

Comment:
I think that the usage of the `context` parameter was inconsistent and confusing.

I rename the top argument top `options` as it's the object which when called from the registry combines the call context, `variables`, loader itself and `basePath`.

Especially that last one was currently assumed to live under the `context` which I think was a mistake due to clashing names.